lib/rpmdb.c

```diff
@@ -2281,7 +2281,7 @@ static int rpmdbRemoveFiles(char * pattern)
     int rc = 0;
     ARGV_t paths = NULL, p;
 
-    if (rpmGlob(pattern, NULL, &paths) == 0) {
+    if (rpmGlob(pattern, RPMGLOB_NONE, NULL, &paths) == 0) {
```

Comment:
Similar to the above, we would try to unlink a file that doesn't exist (e.g. one that ends with a `/*`), see `rpmdbRemoveDatabase()` where we call this function.